py/server/deephaven/updateby.py

```diff
@@ -0,0 +1,276 @@
+        if ema_control is None:
+            return UpdateByClause(j_updateby_clause=_JUpdateByClause.Ema(ts_col, time_scale, *cols))
+        else:
+            return UpdateByClause(
+                j_updateby_clause=_JUpdateByClause.Ema(ema_control.j_ema_control, ts_col, time_scale, *cols))
+    except Exception as e:
+        raise DHError(e, "failed to create a time-decay EMA UpdateByClause.") from e
+
+
+def cum_sum(cols: Union[str, List[str]]) -> UpdateByClause:
+    """Creates a CumSum (cumulative sum) update-by clause for the supplied column names.
+
+    Args:
+
+        cols (Union[str, List[str]]): the column(s) to be operated on, can be renaming expressions, i.e. "new_col =
+            col"; when empty, update_by performs the cumulative sum operation on all the applicable columns.
+
+    Returns:
+        a UpdateByClause
+
+    Raises:
+        DHError
+    """
+    try:
+        cols = to_sequence(cols)
+        return UpdateByClause(j_updateby_clause=_JUpdateByClause.CumSum(cols))
+    except Exception as e:
+        raise DHError(e, "failed to create a cumulative sum UpdateByClause.") from e
+
+
+def cum_prod(cols: Union[str, List[str]]) -> UpdateByClause:
+    """Creates a CumProd (cumulative product) update-by clause for the supplied column names.
+
+    Args:
+        cols (Union[str, List[str]]): the column(s) to be operated on, can be renaming expressions, i.e. "new_col =
+            col"; when empty, update_by perform the cumulative produce operation on all the applicable columns.
+
+    Returns:
+        a UpdateByClause
+
+    Raises:
+        DHError
+    """
+    try:
+        cols = to_sequence(cols)
+        return UpdateByClause(j_updateby_clause=_JUpdateByClause.CumProd(cols))
+    except Exception as e:
+        raise DHError(e, "failed to create a cumulative product UpdateByClause.") from e
+
+
+def cum_min(cols: Union[str, List[str]]) -> UpdateByClause:
+    """Creates a CumMin (cumulative minimum) update-by clause for the supplied column names.
+
+    Args:
+        cols (Union[str, List[str]]): the column(s) to be operated on, can be renaming expressions, i.e. "new_col =
+            col"; when empty, update_by perform the cumulative minimum operation on all the applicable columns.
+
+    Returns:
+        a UpdateByClause
+
+    Raises:
+        DHError
+    """
+    try:
+        cols = to_sequence(cols)
+        return UpdateByClause(j_updateby_clause=_JUpdateByClause.CumMin(cols))
+    except Exception as e:
+        raise DHError(e, "failed to create a cumulative minimum UpdateByClause.") from e
+
+
+def cum_max(cols: Union[str, List[str]]) -> UpdateByClause:
+    """Creates a CumMax (cumulative maximum) update-by clause for the supplied column names.
+
+    Args:
+        cols (Union[str, List[str]]): the column(s) to be operated on, can be renaming expressions, i.e. "new_col = 
+            col"; when empty, update_by perform the cumulative maximum operation on all the applicable columns.
```

Comment:
```suggestion
        cols (Union[str, List[str]]): the column(s) to be operated on, can include expressions to rename the output, i.e. "new_col = 
            col"; when empty, update_by performs the cumulative maximum operation on all the applicable columns.
```
Duplicated in many places.